### ISSUE TITLE
Avoid removing invalid value for relations

### ISSUE THREAD

**Issue by OlafvdSpek** [phpmyadmin/phpmyadmin#12115]
I've got a MyISAM table with a 'foreign' key / pMA relation defined into another table. 
When editing a row with a key value that's NOT present in the 'target' table, the dropdown value is blank. Submitting the form results in the value being reset to 0 / default.

1. pMA should NOT reset the value.
2. pMA should allow me to enter an orphaned value, which can't be done with the dropdown.

pMA 4.5.5.1deb1

**Comment by madhuracj** [comment]
Issue 2 has already been requested in the past in https://github.com/phpmyadmin/phpmyadmin/issues/11129 and rejected.

**Comment by nijel** [comment]
I really don't think we should support this. It would just complicate the UI with no clear benefits.

**Comment by OlafvdSpek** [comment]
No benefits?
pMA is messing up our data...

Only allowing entry via a dropdown also means I can't paste in a value.

**Comment by nijel** [comment]
Your data is already messed up in this case. If you want arbitrary values I'd recommend you not to use relations at all.

**Comment by OlafvdSpek** [comment]
Our data was fine until pMA updated it... ;)

Not using relations would be an option but why can't we take advantage of teh other benefits it provides?

**Comment by nijel** [comment]
As we want to provide same interface for both MySQL native and phpMyAdmin internal relations, adding free form input really doesn't make sense as that is not possible with native relations as the target value has to exist. Actually I don't see point in using relations then...

**Comment by OlafvdSpek** [comment]
At least detect the problem and refuse the row to be edited, but DO NOT silently change the value.

The benefit relations provide is the values being links to the foreign table.

**Comment by OlafvdSpek** [comment]
Could priority / status of this issue be reconsidered?
Silent data corruption is a high-priority bug.
Is there a setting to disable the dropdown entirely?

**Comment by nijel** [comment]
Yes, don't use built in relations. There is no separate option for dropdowns.

**Comment by OlafvdSpek** [comment]
But I like relations..
I just hate the dropdown.

**Comment by nijel** [comment]
Well you only like parts of the relations, for example not the most important thing to have consistent data in the tables....

**Comment by OlafvdSpek** [comment]
As I said before my data is (was? ;) ) just fine.

**Comment by nijel** [comment]
Best approach to address this probably would be to change PMA_getForeignData in libraries/relation.lib.php so that it always includes current value in disp_row results...

**Comment by OlafvdSpek** [comment]
Sounds like a plan. ;)
